apps/10address-book-ajax/script.js

```diff
@@ -1,5 +1,129 @@
 'use strict';
+var sortType = "id";
+var reverseSort = false;
 
 $(document).ready(function() {
-  // You code here
+  queryPage(sortType)
+
+  //Create a method that animates a div and adds a listener to remove the animation class when finished.
+  $.fn.extend({
+        animateCss: function (animationName) {
+            var animationEnd = 'webkitAnimationEnd mozAnimationEnd MSAnimationEnd oanimationend animationend';
+            this.addClass('animated ' + animationName).one(animationEnd, function() {
+                $(this).removeClass('animated ' + animationName);
+            });
+        }
+    });
+
+  //Handle clicks of links to individual users.
+  $('body').on("click", "tr td a", function(event) {
+    event.preventDefault();
+    var theID=$(this).data('id');
+    var theURL='https://reqres-api.herokuapp.com/api/users' + '/' + theID;
+
+    $('#details-wrapper').animateCss('bounceInRight');
+
+    $.ajax(theURL, {
+        success: function(currentUser) {
+            var firstName = currentUser.first_name;
+            var lastName = currentUser.last_name;
+            var studentID = currentUser.id;
+            var address = currentUser.address;
+            var phone = currentUser.phone;
+            var occupation = currentUser.occupation;
+            var avatarAddress = currentUser.avatar;
+
+            var theURL = 'https://reqres-api.herokuapp.com/api/users' + '/' + studentID;
+
+            var theHTML=
+            '<div id="details">' +
+              '<h3>' + firstName + ' ' + lastName + '</h3>' +
+              '<h4>' + occupation + '</h4>' +
+              '<p>' + phone + '</p>' +
+              '<p>' + address + '</p>' +
+              '<img src="' + avatarAddress + '">' +
+            '</div>';
+
+            $('#details').replaceWith(theHTML)
+        }
+      });
+    });
+
+  //Handle sorting when a sort link is pressed.
+  $('body').on("click", "a.sort", function() {
+    if ($(this).hasClass("id")) {
+      (sortType === "id") ? ((reverseSort) ? reverseSort = false : reverseSort = true) : reverseSort = false;
+      sortType = "id";
+    }
+
+    if ($(this).hasClass("fname")) {
+      (sortType === "fname") ? ((reverseSort) ? reverseSort = false : reverseSort = true) : reverseSort = false;
+      sortType = "fname";
+    }
+
+    if ($(this).hasClass("lname")) {
+      (sortType === "lname") ? ((reverseSort) ? reverseSort = false : reverseSort = true) : reverseSort = false;
+      sortType = "lname";
+    }
+
+    queryPage(sortType)
+  });
+
 });
+
+function queryPage(theSort) {
+  //Build the initial data.
+  $.ajax('https://reqres-api.herokuapp.com/api/users', {
+    success: function(users) {
+      $('tbody').empty();
+      var theHTML = "";
+
+      switch (sortType) {
+        case "id":
+          users.sort(function(a,b) {
```

Comment:
Since the `"id"` attribute is a number you can can call sort on them directly.

Ex: `users.sort(function(a, b) { return b.id - a.id; });`